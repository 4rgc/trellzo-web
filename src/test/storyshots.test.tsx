import initStoryshots from '@storybook/addon-storyshots';
import { render } from '@testing-library/react';

const reactTestingLibrarySerializer = {
	print: (val: any, serialize: any, indent: any) =>
		// eslint-disable-next-line testing-library/no-node-access
		serialize(val.container.firstChild),
	test: (val: any) => val && val.hasOwnProperty('container'),
};

initStoryshots({
	renderer: render,
	snapshotSerializers: [reactTestingLibrarySerializer],
});
