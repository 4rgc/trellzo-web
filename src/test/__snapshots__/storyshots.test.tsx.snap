// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Button Custom Class 1`] = `
<button
  className="btn-prm-md-reg btn-dis"
>
  fake disabled
</button>
`;

exports[`Storyshots Components/Button Default 1`] = `
<button
  className="btn-prm-md-reg "
/>
`;

exports[`Storyshots Components/Button Disabled 1`] = `
<button
  className="btn-prm-dis-md-reg "
>
  disabled
</button>
`;

exports[`Storyshots Components/Button Large 1`] = `
<button
  className="btn-prm-lg-reg "
>
  Large Button !!!
</button>
`;

exports[`Storyshots Components/Button Medium 1`] = `
<button
  className="btn-prm-md-reg "
>
  Medium Btn
</button>
`;

exports[`Storyshots Components/Button Primary 1`] = `
<button
  className="btn-prm-md-reg "
>
  Primary
</button>
`;

exports[`Storyshots Components/Button Secondary 1`] = `
<button
  className="btn-sec-md-reg "
>
  Secondary
</button>
`;

exports[`Storyshots Components/Button Small 1`] = `
<button
  className="btn-prm-sm-reg "
>
  Small
</button>
`;

exports[`Storyshots Components/Button Square 1`] = `
<button
  className="btn-prm-sq-reg "
>
  OK
</button>
`;

exports[`Storyshots Components/Button Thin 1`] = `
<button
  className="btn-prm-md-thn "
/>
`;

exports[`Storyshots Components/Card Default 1`] = `
<div
  className="card card-md"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <svg
    className="card-cover"
    fill="black"
    preserveAspectRatio="xMinYMin slice"
    stroke="black"
  >
    placeholder.svg
  </svg>
  <div
    className="card-divider"
  />
  <div
    className="card-container"
  >
    <h3
      className="card-title"
    >
      Title
    </h3>
    <div
      className="card-content-text"
    >
      Labore odio voluptas voluptas vel qui vel. Aut est quo. Eveniet non tempore incidunt eum vero. Quia tempore error neque nam et numquam quidem. Amet voluptas delectus quisquam architecto numquam quia et impedit eos. Excepturi aliquid eum.
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/Card With Cover 1`] = `
<div
  className="card card-md"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <img
    alt="card img"
    className="card-cover"
    src="/regular-cover.png"
  />
  <div
    className="card-divider"
  />
  <div
    className="card-container"
  >
    <h3
      className="card-title"
    >
      Title
    </h3>
    <div
      className="card-content-text"
    >
      Labore odio voluptas voluptas vel qui vel. Aut est quo. Eveniet non tempore incidunt eum vero. Quia tempore error neque nam et numquam quidem. Amet voluptas delectus quisquam architecto numquam quia et impedit eos. Excepturi aliquid eum.
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/Card With Custom Children 1`] = `
<div
  className="card card-md"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <svg
    className="card-cover"
    fill="black"
    preserveAspectRatio="xMinYMin slice"
    stroke="black"
  >
    placeholder.svg
  </svg>
  <div
    className="card-divider"
  />
  <div
    className="card-container"
  >
    <div
      style={
        Object {
          "backgroundColor": "blue",
          "borderRadius": "10px",
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      Custom stuff
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/Card With Custom Content 1`] = `
<div
  className="card card-md"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <svg
    className="card-cover"
    fill="black"
    preserveAspectRatio="xMinYMin slice"
    stroke="black"
  >
    placeholder.svg
  </svg>
  <div
    className="card-divider"
  />
  <div
    className="card-container-no-text"
  >
    <h3
      className="card-title"
    >
      Title
    </h3>
    <div
      className=""
    >
      <div
        style={
          Object {
            "backgroundColor": "blue",
            "borderRadius": "10px",
            "color": "white",
            "textAlign": "center",
          }
        }
      >
        Custom content
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/Card With Horizontal Cover 1`] = `
<div
  className="card card-md"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <img
    alt="card img"
    className="card-cover"
    src="/horizontal-cover.png"
  />
  <div
    className="card-divider"
  />
  <div
    className="card-container"
  >
    <h3
      className="card-title"
    >
      Title
    </h3>
    <div
      className="card-content-text"
    >
      Labore odio voluptas voluptas vel qui vel. Aut est quo. Eveniet non tempore incidunt eum vero. Quia tempore error neque nam et numquam quidem. Amet voluptas delectus quisquam architecto numquam quia et impedit eos. Excepturi aliquid eum.
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/Card With Long Title 1`] = `
<div
  className="card card-md"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <svg
    className="card-cover"
    fill="black"
    preserveAspectRatio="xMinYMin slice"
    stroke="black"
  >
    placeholder.svg
  </svg>
  <div
    className="card-divider"
  />
  <div
    className="card-container"
  >
    <h3
      className="card-title"
    >
      Hello this is long title
    </h3>
    <div
      className="card-content-text"
    >
      Labore odio voluptas voluptas vel qui vel. Aut est quo. Eveniet non tempore incidunt eum vero. Quia tempore error neque nam et numquam quidem. Amet voluptas delectus quisquam architecto numquam quia et impedit eos. Excepturi aliquid eum.
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/Card With Vertical Cover 1`] = `
<div
  className="card card-md"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <img
    alt="card img"
    className="card-cover"
    src="/vertical-cover.png"
  />
  <div
    className="card-divider"
  />
  <div
    className="card-container"
  >
    <h3
      className="card-title"
    >
      Title
    </h3>
    <div
      className="card-content-text"
    >
      Labore odio voluptas voluptas vel qui vel. Aut est quo. Eveniet non tempore incidunt eum vero. Quia tempore error neque nam et numquam quidem. Amet voluptas delectus quisquam architecto numquam quia et impedit eos. Excepturi aliquid eum.
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard Default 1`] = `
<div
  className="card card-md card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={14}
      value=""
    >
      
    </textarea>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard Large 1`] = `
<div
  className="card card-lg card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={26}
      value=""
    >
      
    </textarea>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard Large With Title 1`] = `
<div
  className="card card-lg card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <input
      className="tinput-md title"
      onChange={[Function]}
      value=""
    />
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={24}
      value=""
    >
      
    </textarea>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard Medium 1`] = `
<div
  className="card card-md card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={14}
      value=""
    >
      
    </textarea>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard Small 1`] = `
<div
  className="card card-sm card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={3}
      value=""
    >
      
    </textarea>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard Small With A Button 1`] = `
<div
  className="card card-sm card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={1}
      value=""
    >
      
    </textarea>
    <div
      className="btn-container"
    >
      <button
        className="btn-prm-md-thn "
        onClick={[Function]}
      >
        click me
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard Small With Title 1`] = `
<div
  className="card card-sm card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={3}
      value=""
    >
      
    </textarea>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard With A Button 1`] = `
<div
  className="card card-md card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={12}
      value=""
    >
      
    </textarea>
    <div
      className="btn-container"
    >
      <button
        className="btn-prm-md-thn "
        onClick={[Function]}
      >
        click me
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard With Title 1`] = `
<div
  className="card card-md card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <input
      className="tinput-md title"
      onChange={[Function]}
      value=""
    />
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={12}
      value=""
    >
      
    </textarea>
  </div>
</div>
`;

exports[`Storyshots Components/EditableCard With Two Buttons 1`] = `
<div
  className="card card-md card-noimage editable-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <textarea
      className="tinput-md"
      onChange={[Function]}
      placeholder=""
      rows={12}
      value=""
    >
      
    </textarea>
    <div
      className="btn-container"
    >
      <button
        className="btn-sec-sm-thn "
        onClick={[Function]}
      >
        cancel
      </button>
      <button
        className="btn-prm-sm-thn "
        onClick={[Function]}
      >
        ok
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/GhostCard Default 1`] = `
<div
  className="card card-md ghost-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <svg
    className="card-cover"
    fill="black"
    preserveAspectRatio="xMinYMin slice"
    stroke="black"
  >
    placeholder.svg
  </svg>
  <div
    className="card-divider"
  />
  <div
    className="card-container-no-text"
  >
    <div
      className="ghost-card-text-container"
    >
      <b
        className="ghost-card-text"
      >
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/GhostCard Large 1`] = `
<div
  className="card card-lg ghost-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <svg
    className="card-cover"
    fill="black"
    preserveAspectRatio="xMinYMin slice"
    stroke="black"
  >
    placeholder.svg
  </svg>
  <div
    className="card-divider"
  />
  <div
    className="card-container-no-text"
  >
    <div
      className="ghost-card-text-container"
    >
      <b
        className="ghost-card-text"
      >
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/GhostCard No Image 1`] = `
<div
  className="card card-md card-noimage ghost-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <div
    className="card-container-no-text"
  >
    <div
      className="ghost-card-text-container"
    >
      <b
        className="ghost-card-text"
      >
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/GhostCard Small 1`] = `
<div
  className="card card-sm card-noimage ghost-card"
  style={
    Object {
      "width": "200px",
    }
  }
>
  <svg
    className="card-cover"
    fill="black"
    preserveAspectRatio="xMinYMin slice"
    stroke="black"
  >
    placeholder.svg
  </svg>
  <div
    className="card-divider"
  />
  <div
    className="card-container-no-text"
  >
    <div
      className="ghost-card-text-container"
    >
      <b
        className="ghost-card-text"
      >
        Add a new asdf...
      </b>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/NotesList Default 1`] = `
<div
  className="notes-list"
>
  <span
    className="notes-list-name"
  >
    List 1
  </span>
  <div
    className="card card-sm card-noimage"
  >
    <div
      className="card-container"
    >
      <h3
        className="card-title"
      >
        Note Something
      </h3>
      <div
        className="card-content-text"
      >
        Buy something
      </div>
    </div>
  </div>
  <div
    className="card card-sm card-noimage"
  >
    <div
      className="card-container"
    >
      <h3
        className="card-title"
      >
        Delete me
      </h3>
      <div
        className="card-content-text"
      >
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Components/NotesList Empty 1`] = `
<div
  className="notes-list"
>
  <span
    className="notes-list-name"
  >
    List aa
  </span>
  <span
    className="notes-list-empty-msg"
  >
    Nothing here yet.
  </span>
</div>
`;

exports[`Storyshots Components/TextInput Bigger Multiline 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "20px",
      "width": "400px",
    }
  }
>
  <textarea
    className="tinput-md"
    onChange={[Function]}
    rows={6}
    value=""
  >
    
  </textarea>
</div>
`;

exports[`Storyshots Components/TextInput Default 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "20px",
      "width": "400px",
    }
  }
>
  <input
    className="tinput-md"
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`Storyshots Components/TextInput In A Tight Container 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "10px",
      "width": "100px",
    }
  }
>
  <input
    className="tinput-md"
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`Storyshots Components/TextInput Large Font 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "20px",
      "width": "400px",
    }
  }
>
  <input
    className="tinput-lg"
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`Storyshots Components/TextInput Multiline 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "20px",
      "width": "400px",
    }
  }
>
  <textarea
    className="tinput-md"
    onChange={[Function]}
    value=""
  >
    
  </textarea>
</div>
`;

exports[`Storyshots Components/TextInput Multiline With Placeholder 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "20px",
      "width": "400px",
    }
  }
>
  <textarea
    className="tinput-md"
    onChange={[Function]}
    placeholder="Type something long here..."
    value=""
  >
    
  </textarea>
</div>
`;

exports[`Storyshots Components/TextInput Single Line With Rows Specified 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "20px",
      "width": "400px",
    }
  }
>
  <input
    className="tinput-md"
    onChange={[Function]}
    rows={5}
    value=""
  />
</div>
`;

exports[`Storyshots Components/TextInput With Placeholder 1`] = `
<div
  style={
    Object {
      "border": "2px black solid",
      "padding": "20px",
      "width": "400px",
    }
  }
>
  <input
    className="tinput-md"
    onChange={[Function]}
    placeholder="Enter something..."
    value=""
  />
</div>
`;

exports[`Storyshots Components/WarningFab Always Display 1`] = `
<div
  className="warningFab"
  style={
    Object {
      "display": "block",
    }
  }
>
  <img
    alt="warning icon"
    className="icon"
    onClick={[Function]}
    src="warning.svg"
  />
</div>
`;

exports[`Storyshots Components/WarningFab Default 1`] = `
<div
  className="warningFab"
  style={
    Object {
      "display": "none",
    }
  }
>
  <img
    alt="warning icon"
    className="icon"
    onClick={[Function]}
    src="warning.svg"
  />
</div>
`;

exports[`Storyshots Components/WarningFab Display On Message 1`] = `
<div
  className="warningFab"
  style={
    Object {
      "display": "block",
    }
  }
>
  <img
    alt="warning icon"
    className="icon"
    onClick={[Function]}
    src="warning.svg"
  />
</div>
`;

exports[`Storyshots Components/WarningFab Display On Message No Message 1`] = `
<div
  className="warningFab"
  style={
    Object {
      "display": "none",
    }
  }
>
  <img
    alt="warning icon"
    className="icon"
    onClick={[Function]}
    src="warning.svg"
  />
</div>
`;

exports[`Storyshots Components/WarningFab With Custom Action 1`] = `
<div
  className="warningFab"
  style={
    Object {
      "display": "block",
    }
  }
>
  <img
    alt="warning icon"
    className="icon"
    onClick={[Function]}
    src="warning.svg"
  />
</div>
`;
